src/raftstore/store/peer.rs

```diff
@@ -353,6 +358,52 @@ impl Peer {
         Ok(())
     }
 
+    pub fn step(&mut self, m: eraftpb::Message) -> Result<()> {
+        if self.is_leader() && m.get_from() != INVALID_ID {
+            self.peer_heartbeats.insert(m.get_from(), Instant::now());
+        }
+        try!(self.raft_group.step(m));
+        Ok(())
+    }
+
+    pub fn check_peers(&mut self) {
+        if !self.is_leader() {
+            self.peer_heartbeats.clear();
+            return;
+        }
+
+        let peers = self.get_store().get_region().get_peers().to_owned();
```

Comment:
Since peer don't know when it becomes leader, I still need to do some check here.
I optimize the check to make it more light weight.